libs/wingsdk/src/target-tf-aws/service.ts

```diff
@@ -0,0 +1,338 @@
+    props: cloud.ServiceProps = {}
+  ) {
+    super(scope, id, handler, props);
+    // We probably need to create the ECR per app (App.of(this).ecr)
+
+    this.workdir = App.of(this).workdir;
+    this.wrapperEntrypoint = join(this.workdir, `${this.assetName}_wrapper.js`);
+    this.dockerFileName = `Dockerfile_${this.assetName}`;
+    let app = App.of(this) as App;
+
+    // This forces the lazy loading of the properties (TODO: find a better way to do this)
+    app.ecrAuth;
+    app.dockerProvider;
+    app.vpc;
+    
+    let image = new Image(this, "DockerImage", {
+      name: `${app.ecr.repositoryUrl}:${this.assetName}`,
+      buildAttribute: {
+        context: ".wing",
+        dockerfile: this.dockerFileName,
+        platform: "linux/amd64",
+      }
+    });
+
+    new RegistryImage(this, "RegistryImage", {
+      name: image.name
+    });
+
+    let executionRole = new IamRole(this, "ExecutionRole", {
+      assumeRolePolicy: JSON.stringify({
+        Version: "2012-10-17",
+        Statement: [
+          {
+            Effect: "Allow",
+            Principal: {
+              Service: "ecs-tasks.amazonaws.com",
+            },
+            Action: "sts:AssumeRole",
+          },
+        ],
+      }),
+      inlinePolicy: [{
+        name: `${this.assetName}-inline-policy`,
+        policy: JSON.stringify({
+          Version: "2012-10-17",
+          Statement: [
+            {
+              "Effect": "Allow",
+              "Action": [
+                "logs:CreateLogStream",
+                "logs:PutLogEvents",
+                "logs:CreateLogGroup"
+              ],
+              "Resource": "*"
+            },
+            {
+              "Effect": "Allow",
+              "Action": [
+                "ecr:BatchGetImage",
+                "ecr:GetDownloadUrlForLayer",
+                "ecr:GetAuthorizationToken"
+              ],
+              "Resource": "*"
+            }
+          ],
+        }),
+      }]
+    });
+
+    let taskRole = new IamRole(this, "TaskRole", {
+      assumeRolePolicy: JSON.stringify({
+        Version: "2012-10-17",
+        Statement: [
+          {
+            Effect: "Allow",
+            Principal: {
+              Service: "ecs-tasks.amazonaws.com",
+            },
+            Action: "sts:AssumeRole",
+          },
+        ],
+      }),
+      inlinePolicy: [{
+        name: `${this.assetName}-inline-policy`,
+        policy: Lazy.stringValue({
+          produce: () => {
+            this.policyStatments = this.policyStatments ?? [];
+            
+            if (this.policyStatments.length !== 0) {
+              return JSON.stringify({
+                Version: "2012-10-17",
+                Statement: this.policyStatments
+              });
+            }
+
+            return JSON.stringify({
+              Version: "2012-10-17",
+              Statement: [
+                {
+                  Effect: "Allow",
+                  Action: "none:null",
```

Comment:
Do we really need this weird null statement? Can't we just create a policy without statements?

Reply:
Im not sure we do this same thing for `function.ts` with a comment that says: 
```
// policy must contain at least one statement, so include a no-op statement
```

So I assume its needed, but I can test